server/internal/infrastructure/memory/model_test.go

```diff
@@ -0,0 +1,251 @@
+		f:    repo.WorkspaceFilter{},
+	}
+	r.data.Store(mId, model.New().NewID().Schema(id.NewSchemaID()).RandomKey().Project(pId).MustBuild())
+
+	got, err := r.CountByProject(ctx, pId)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, got)
+
+	err = errors.New("test")
+	r = &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    repo.WorkspaceFilter{},
+		err:  err,
+	}
+	got, gotErr := r.CountByProject(ctx, pId)
+	assert.Equal(t, gotErr, err)
+	assert.Equal(t, 0, got)
+}
+
+func TestMemory_FindByKey(t *testing.T) {
+	ctx := context.Background()
+	pId := id.NewProjectID()
+	mId := id.NewModelID()
+	key := key.Random()
+
+	r := &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    repo.WorkspaceFilter{},
+	}
+	r.data.Store(mId, model.New().NewID().Schema(id.NewSchemaID()).Key(key).Project(pId).MustBuild())
+	expected := r.data.Find(func(_ id.ModelID, m *model.Model) bool {
+		return m.Key().String() == key.String() && m.Project() == pId
+	})
+	got, err := r.FindByKey(ctx, pId, key.String())
+	assert.Nil(t, err)
+	assert.Equal(t, expected, got)
+
+	r = &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    repo.WorkspaceFilter{},
+	}
+	got, gotErr := r.FindByKey(ctx, pId, key.String())
+	assert.Equal(t, gotErr, rerror.ErrNotFound)
+	assert.Nil(t, got)
+
+	err = errors.New("test")
+	r = &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    repo.WorkspaceFilter{},
+		err:  err,
+	}
+	got, gotErr = r.FindByKey(ctx, pId, key.String())
+	assert.Nil(t, got)
+	assert.Equal(t, gotErr, errors.New("test"))
+}
+
+func TestMemory_FindByID(t *testing.T) {
+	ctx := context.Background()
+	pId := id.NewProjectID()
+	mId := id.NewModelID()
+	key := key.Random()
+	r := &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    repo.WorkspaceFilter{},
+	}
+	got, err := r.FindByID(ctx, mId)
+	assert.Nil(t, got)
+	assert.ErrorIs(t, err, rerror.ErrNotFound)
+
+	r.data.Store(mId, model.New().NewID().Schema(id.NewSchemaID()).Key(key).Project(pId).MustBuild())
+	expected := r.data.Find(func(k id.ModelID, m *model.Model) bool {
+		return k == mId
+	})
+	got, err = r.FindByID(ctx, mId)
+	assert.Nil(t, err)
+	assert.Equal(t, expected, got)
+
+	err = errors.New("test")
+	r = &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    repo.WorkspaceFilter{},
+		err:  err,
+	}
+	got, gotErr := r.FindByID(ctx, mId)
+	assert.Equal(t, gotErr, err)
+	assert.Nil(t, got)
+}
+
+func TestMemory_FindByIDs(t *testing.T) {
+	ctx := context.Background()
+	pId := id.NewProjectID()
+	mId := id.NewModelID()
+	key := key.Random()
+	mIds := id.ModelIDList{}
+	r := &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    repo.WorkspaceFilter{},
+	}
+	r.data.Store(mId, model.New().NewID().Schema(id.NewSchemaID()).Key(key).Project(pId).MustBuild())
+
+	expectedModelList := model.List(r.data.FindAll(func(k id.ModelID, m *model.Model) bool {
+		return mIds.Has(k)
+	})).SortByID()
```

Comment:
```suggestion
	seed :=  model.New().NewID().Schema(id.NewSchemaID()).Key(key).Project(pId).MustBuild()
	r.data.Store(mId, seed)

	expectedModelList := model.List(seed)
```

Do not aim excessively for DRY in test codes, but give priority to simple, readable code. An expected value is good to be written manually.

Fix others also